src/daft-catalog/src/session.rs

```diff
@@ -0,0 +1,51 @@
+use daft_logical_plan::LogicalPlanBuilder;
+
+use crate::{errors::Result, identifier::Identifier, DaftCatalog};
+
+/// Temporary rename while refactoring.
+type Metastore = DaftCatalog;
+
+/// Session holds all state for query planning and execution (e.g. connection).
+pub struct Session {
+    /// Session identifier
+    pub id: String,
+    // TODO remove DaftCatalog after all APIs are migrated.
+    pub metastore: DaftCatalog,
+    // TODO execution context
+    // TODO session options
+}
+
+impl Session {
+    /// Creates a new empty session
+    pub fn new(id: &str, metastore: Metastore) -> Self {
+        Self {
+            id: id.to_string(),
+            metastore,
+        }
+    }
+
+    /// Creates a table backed by the view
+    /// TODO support names/namespaces and other table sources.
+    pub fn create_table(&mut self, name: &str, view: impl Into<LogicalPlanBuilder>) -> Result<()> {
```

Comment:
I'd rather lock the metastore and other state behind a private state struct with inner mutability. 

something like

```rs

struct SessionState {
  metastore: DaftCatalog
  // ..
}

pub struct Session {
  state: Arc<RwLock<SessionState>>
  // ...
}
impl Session {
  pub fn create_table(&self, ...)
}
```

That way we can just pass around `Session` or `Arc<Session>` instead of needing the user of the session externally manage mutability.